src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs

```diff
@@ -614,10 +614,11 @@ private SingleNamespaceOrTypeDeclaration VisitTypeDeclaration(TypeDeclarationSyn
             }
 
             // A type with parameters at least has a primary constructor
-            bool hasPrimaryCtor = node is TypeDeclarationSyntax { ParameterList: { } };
+            var hasPrimaryCtor = node.ParameterList != null;
             if (hasPrimaryCtor)
```

Comment:
>if (hasPrimaryCtor) [](http://example.com/codeflow?start=12&length=19)

Probably shouldn't enter this ```if``` for interfaces.

Reply:
I will clean this up separately.

Reply:
sgtm.